### PR TITLE
fix: Fixed oauth/token endpoint to accept x-www-form-urlencoded

### PR DESCRIPTION
**Pull request by JigarJoshi**
Fixed oauth/token endpoint to accept application/x-www-form-urlencoded payload

### CODE REVIEW

File: api/server/v1/marshaler.go [hunk 1]
```diff
@@ -507,10 +575,10 @@ func (x *FacetStats) MarshalJSON() ([]byte, error) {
...
 		Count int64    `json:"count"`
 	}{
-		Avg:   x.Avg,
-		Max:   x.Max,
-		Min:   x.Min,
-		Sum:   x.Sum,
+		Avg:   &x.Avg,
+		Max:   &x.Max,
+		Min:   &x.Min,
+		Sum:   &x.Sum,
```
Comment (JigarJoshi) [inline api/server/v1/marshaler.go:581]: this is not related change.

File: api/server/v1/marshaler.go [hunk 2]
```diff
@@ -16,18 +16,61 @@ package api
...
+		}
+	}
+	return f.jsonDecoder.Decode(dst)
+}
+
 // CustomMarshaler is a marshaler to customize the response. Currently, it is only used to marshal custom error message
 // otherwise it just uses the inbuilt mux marshaller.
 type CustomMarshaler struct {
-	*runtime.JSONBuiltin
+	JsonBuiltin *runtime.JSONBuiltin
```
Comment (efirs) [inline api/server/v1/marshaler.go:57]: We can keep embedded version or we should rename the member to JSONBuiltin otherwise linters would complain. It expectes JSON to be capitalized.

Comment (JigarJoshi) [reply api/server/v1/marshaler.go]: updated the variable name and keeping the variable here to be clear explicitly on how the methods are implemented.

**Review by efirs — APPROVED** [review]

File: api/server/v1/marshaler.go [hunk 3]
```diff
@@ -53,6 +96,31 @@ func (c *CustomMarshaler) Marshal(v interface{}) ([]byte, error) {
 	return c.JSONBuiltin.Marshal(v)
 }
 
+func (c *CustomMarshaler) Unmarshal(data []byte, v interface{}) error {
+	switch v.(type) {
+	case *GetAccessTokenRequest:
+		{
```
Comment (himank) [inline api/server/v1/marshaler.go:102]: inside the case, you don't need any braces i.e. 
```
switch v.(type) {
	case *GetAccessTokenRequest:
	    return unmarshalInternal(data, v)
}
```
